### PR TITLE
FEAT: Refactor outport driver to work with proto messages

### PR DESCRIPTION
**Pull request by mariusmihaic**
Refactored `Driver` interface to accept only proto defined structs
- Deleted every old struct from `shared.go` and redefined them as proto messages
- When reviewing this PR, one should carefully check that the old structs from `shared.go` (e.g.: `ArgsSaveRoundsInfo`) have been correctly replaced with the ones defined in `outportBlock.proto`

### CODE REVIEW

File: data/outport/outportBlock.proto [hunk 1]
```diff
@@ -80,11 +80,39 @@ message ValidatorRatingInfo {
...
+}
+
 message RoundInfo {
-  uint64 Index                   = 1 [(gogoproto.jsontag) = "index"];
-  repeated uint64 SignersIndexes = 2 [(gogoproto.jsontag) = "signersIndexes"];
-  bool BlockWasProposed          = 3 [(gogoproto.jsontag) = "blockWasProposed"];
-  uint32 ShardId                 = 4 [(gogoproto.jsontag) = "shardId"];
-  uint32 Epoch                   = 5 [(gogoproto.jsontag) = "epoch"];
-  uint64 Timestamp               = 6 [(gogoproto.jsontag) = "timestamp"];
+  uint64          Index            = 1 [(gogoproto.jsontag) = "index"];
```
Comment (miiu96) [inline data/outport/outportBlock.proto:89]: this should be renamed in `Round`

File: data/outport/outportBlock.proto [hunk 2]
```diff
@@ -80,11 +80,39 @@ message ValidatorRatingInfo {
...
+
+message ValidatorsPubKeys{
+  map<string, PubKeys> ShardValidatorsPubKeys = 1 [(gogoproto.jsontag) = "validatorsPubKeys,omitempty"];
+  uint32               Epoch                  = 2 [(gogoproto.jsontag) = "epoch"];
+}
+
+message Accounts {
+  uint32 ShardID                              = 1 [(gogoproto.jsontag) = "shardID"];
+  uint64 BlockTimestamp                       = 2 [(gogoproto.jsontag) = "blockTimestamp"];
+  map<string, AlteredAccount> AlteredAccounts = 3 [(gogoproto.jsontag) = "shardID,omitempty"];
```
Comment (miiu96) [inline data/outport/outportBlock.proto:113]: the json tag is wrong here

**Review by miiu96 — CHANGES_REQUESTED** [review]

File: data/outport/outportBlock.proto (hunk 1 above)
Comment (mariusmihaic) [reply data/outport/outportBlock.proto:89]: I've copied the fields as they were. It was `Index`: https://github.com/multiversx/mx-chain-core-go/blob/9b2d95bc7a05933c8be230ad6730faff8b4a0b2f/data/outport/dtos.go#L94

Shall we rename it?

Comment (miiu96) [reply data/outport/outportBlock.proto:89]: yes. it's ok to rename it because the name was not correct.

File: data/outport/outportBlock.proto [hunk 3]
```diff
@@ -80,11 +80,39 @@ message ValidatorRatingInfo {
   float   Rating    = 2 [(gogoproto.jsontag) = "rating"];
 }
 
+message ValidatorsRating{
+  string   IndexID                                   = 1 [(gogoproto.jsontag) = "indexID"];
```
Comment (miiu96) [inline data/outport/outportBlock.proto:84]: this indexID should be break-in 
`ShardID` and `Epoch`

**Review by miiu96 — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**